### PR TITLE
Search Multiple Indices

### PR DESCRIPTION
**Pull request by kyleboe**
https://www.algolia.com/doc/rest-api/search/#search-multiple-indices

I also stubbed out all of the routes for the remainder of the Search Rest API. We will just need to implement the functionality in the future.

### CODE REVIEW

File: lib/algoliax/indexer.ex [hunk 1]
```diff
@@ -132,19 +132,95 @@ defmodule Algoliax.Indexer do
...
+                "params" => "query=bar",
+                "processingTimeMS" => 1,
+                "query" => "bar"
+              }
+            ]
+          }
+        }
+      }
+  """
+  @callback search_multiple(queries :: list(map()), strategy :: String.t()) ::
```
Comment (bronsonholden) [inline lib/algoliax/indexer.ex]: Should this be attached to an Algolia resource at all? Why not just a generic submodule under `Algoliax`? We’ll be using this to query e.g. playlists alongside recent songs, which will have separate Algoliax definitions

Comment (kyleboe) [reply lib/algoliax/indexer.ex]: Good call. I'll move it out from here.

**Review by bronsonholden — APPROVED** [review]
I see green! LGTM 🙌

Out of curiosity, did you exhaustively list paths since we may need to further extend the lib? In particular I'm thinking of analytics endpoints so we can get popular tags so we want to make sure we get those in (at least eventually)

**Comment by kyleboe** [comment]
I just added routing for the all the endpoints on the REST API Search page [here](https://www.algolia.com/doc/rest-api/search/). If we want to do Analytics, we'll have to add the endpoints [here](https://www.algolia.com/doc/rest-api/analytics).

**Pull request merged**